### DISCUSSION TITLE
Changes in vignette on "Perform an integrated analysis"

### DISCUSSION THREAD

**Discussion by RolantusdataExp** [satijalab/seurat#4017] [Q&A]
Hi SatijaLab 
After the recent update of your website, I have noticed that you changed the pibeline for "Perform an integrated analysis" using the NormalizeData + FindVariableFeatures preprocessing. In the old pibeline you did not use the "SelectIntegrationFeatures" before finding anchors, but instead created a refrence.list containing the datasets, which you used as input for FindAnchors + an arguments for dimensions. 
Could you maybe explain the difference between these two approaches in the detection of anchors (Using SelectIntegrationFeatures vs. reference.list)?
I would really appreciate this as it seems to make a significant difference in the performance  

Best regards, 
Peter 
University of Southern Denmark

**Comment by andrewwbutler** [comment]
Hi Peter,

Can you elaborate a bit on the differences are you seeing in terms of performance? From what I understand you are asking about the difference between:

```R
features <- SelectIntegrationFeatures(object.list = object.list)
anchors <- FindIntegrationAnchors(object.list = object.list, anchor.features = features)
```

and 

```R
anchors <- FindIntegrationAnchors(object.list = object.list)
```
In this situation, there shouldn't be any difference in terms of performance. The first chunk is just slightly more explicit about which features are being used. In `FindIntegrationAnchors` if the `anchor.features` parameter isn't specified, the default is "2000". In this scenario, it will then call `SelectIntegrationFeatures` to return 2000 features in the first part of the `FindIntegrationAnchors` function.

**Comment by RolantusdataExp** [comment]
Hi Andrew 
Thank you so much for the reply 
The results are still reproducible between the two pipelines, sorry for the incorrect allegation 
I, however, struggle to reproduce the integration results from my "NormalizeData + FindVariableFeatures preprocessing" when running your SCTransformation pipeline. I do not know if it is expected that the output is different between the two pipelines or maybe you have some tips to obtain the same results? 
Kindest regards, 
Peter

**Reply by andrewwbutler** [reply]
I'm not exactly sure which two pipelines you are comparing. Could you post some example code?

**Reply by RolantusdataExp** [reply]
I am using the pibelines provided in you vignettes :-)

**Reply by andrewwbutler** [reply]
It's not clear from this statement what you are referring to. 
> "NormalizeData + FindVariableFeatures preprocessing" when running your SCTransformation pipeline" 

Rather than me guessing, could you please make it explicit by posting some example code?

**Comment by behyag** [comment]
hi, 
I'd appreciate the explanation to this Q and also I have two more questions:
1. what is the default normalization method for FindIntegrationAnchors() ? "LogNormalize" or "SCT" if it's nit set as in the guidelines?  
2.  I have two datasets that I want to integrate (intact & treated) I have split them in the first part of integration by "condition" which is already added to the metadata of my merged object: 
  condition.list <- SplitObject(my_merged_obj, split.by = "condition")
is is possible to use the "intact" list /dataset as the reference for FindIntegrationAnchors() ? if yes, how should I set this parameter of the function? reference = ?? 

thank you for your help! 
/Behzad

**Reply by timoast** [reply]
Please open a new discussion with your questions, rather than posting them as an answer to a different question
